packages/es-client/indexer.js

```diff
@@ -358,15 +358,15 @@ async function upsertGranule({
     // undesired behavior via business logic on the message write logic
     inline = `
     if ((ctx._source.createdAt === null || params.doc.createdAt >= ctx._source.createdAt)
-      && (params.doc.status != 'running' || (params.doc.status == 'running' && params.doc.execution != ctx._source.execution))) {
+      && ((params.doc.status != 'running' && params.doc.status != 'queued') || ((params.doc.status == 'running' || params.doc.status == 'queued') && params.doc.execution != ctx._source.execution))) {
```

Comment:
So if this is a framework write (write constraints are applicable), the status is not running or queued, OR the status is running or queued but the execution is not the ctx._source.execution?

I think I get the first part where we don't want to update if we're "running" or "queued" (right?) but what's up with the second half? I believe it's correct because it's what we had previously and you're just adding a "queued" check. Just curious.

Reply:
Oh I think I see. The tests are helpful. We're saying that we CAN update if the status is running or queued AND it's a different execution.